File: HanYeojin/Assignment2/yeojin/cafe-server/src/main/java/com/yeojin/cafe/controller/OrderController.java
package com.yeojin.cafe.controller;

import com.yeojin.cafe.dto.OrderResponseDto;
import com.yeojin.cafe.dto.OrderRequestDto;
import com.yeojin.cafe.service.OrderService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/orders")
@RequiredArgsConstructor
public class OrderController {
    private final OrderService orderService;

    // 주문 생성 - POST /api/orders
    @PostMapping
    public OrderResponseDto createOrder(@RequestBody OrderRequestDto requestDto) {
        return orderService.createOrder(requestDto);
    }

    // 주문 조회 - GET /api/orders/{orderId}
    @GetMapping("/{orderId}")
    public OrderResponseDto getOrder(@PathVariable Long orderId) {
        return orderService.getOrderById(orderId);
    }
}

File: assignment2/MinSeungho/cafe/src/main/kotlin/com/seungho/cafe/controller/OrderController.kt
package com.seungho.cafe.controller

import com.seungho.cafe.dto.OrderRequest
import com.seungho.cafe.dto.OrderResponse
import com.seungho.cafe.service.OrderService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/orders")
class OrderController (
    private val orderService: OrderService
){
    @PostMapping
    fun placeOrder(@RequestBody request: OrderRequest): ResponseEntity<OrderResponse> {
        val orderResponse = orderService.placeOrder(request)

        return ResponseEntity.status(HttpStatus.CREATED).body(orderResponse)
    }

    @GetMapping("/{orderId}")
    fun getOrderDetails(@PathVariable orderId: Long) : ResponseEntity<OrderResponse> {
        val orderResponse = orderService.getOrderDetails(orderId)
        return ResponseEntity.ok(orderResponse)
    }
}